src/shared/__tests__/utils.js

```diff
@@ -0,0 +1,65 @@
+import { appendInputValue, isPositiveInteger } from '../utils';
+
+/* eslint-disable comma-dangle */
+
+const defaultArgs = {
+  event: {
+    target: {
+      value: '2',
+      max: '12',
+      min: '1'
+    },
+    key: '2'
+  },
+  onChange: jest.fn()
+};
+
+describe('appendInputValue', () => {
+  describe('when value is higher than max', () => {
+    const { event, onChange } = defaultArgs;
+
+    beforeAll(() => {
+      defaultArgs.event.target.value = '20';
+      appendInputValue(event, onChange);
+    });
+
+    it('set value to max', () => {
+      expect(event.target.value).toEqual('12');
+    });
+
+    it('triggers onChange correctly', () => {
+      expect(onChange).toHaveBeenCalled();
+      expect(onChange).toHaveBeenCalledWith(event);
+    });
+  });
+});
+
+describe('isPositiveInteger', () => {
+  it('should not accept e', () => {
```

Comment:
Sounds like you need it.each :)

Reply:
Not sure I understand, do you prefer only one it() with all cases? :)